source/dhtslib/sam.d

```diff
@@ -728,6 +728,118 @@ private char* reverse(char* str)
     return str;
 }
 
+struct SamWriter{
+    /// filename; as usable from D
+    string filename;
+
+    /// filename \0-terminated C string; reference needed to avoid GC reaping result of toStringz when ctor goes out of scope
+    private immutable(char)* fn;
```

Comment:
I am becoming less fond of immutable, but even const has problems. If this is what we've done for rest of the structs okay to keep but let's review and decide whether shoudl remove from all

Reply:
This is also found in the SAMFile struct. This immutable is useful to prevent mutation of the filename so I do not think it is an issue.